lib/index.js

```diff
@@ -42,11 +42,19 @@ var http = require('http');
 var utils = require('./utils');
 var winston = require('winston');
 
-function AppEngine() {
+function AppEngine(options) {
+  this.options_ = options || {};
   this.serializer_ = new net.proto2.contrib.WireSerializer();
   this.cache_ = {};
-  this.logSizeThreshold_ = 1024 * 1024; // 1 MB
-  this.logMaxTimestampDeltaMillis_ = 60000; // 60 seconds
+  this.defaults_ = {
+    logSizeThreshold: 1024 * 1024, // 1 MB
+    logMaxTimestampDeltaMillis: 60000, // 60 seconds
+    loggerMaxFiles: 1,
+    loggerMaxFileSize: 100 * 1024 * 1024, // 100 MB
+    httpAgentMaxSockets: 100
+  };
+  this.logSizeThreshold_ = this.options_.logSizeThreshold || this.defaults_.logSizeThreshold;
```

Comment:
We need a function that  overrides the values of the defaults with the values of the options.

Requiring user to rely on `options_.v || defaults_.v` is not great for readability in the longer term.